rnd/autogpt_builder/src/app/page.tsx

```diff
@@ -123,56 +97,30 @@ const Monitor = () => {
   );
 };
 
-function flowRunFromNodeExecutionResults(
-  nodeExecutionResults: NodeExecutionResult[],
+function flowRunFromExecutionMeta(
+  graphMeta: GraphMeta,
+  executionMeta: ExecutionMeta,
 ): FlowRun {
-  // Determine overall status
   let status: "running" | "waiting" | "success" | "failed" = "success";
-  for (const execution of nodeExecutionResults) {
-    if (execution.status === "FAILED") {
-      status = "failed";
-      break;
-    } else if (["QUEUED", "RUNNING"].includes(execution.status)) {
-      status = "running";
-      break;
-    } else if (execution.status === "INCOMPLETE") {
-      status = "waiting";
-    }
+  if (executionMeta.status === "FAILED") {
+    status = "failed";
+  } else if (["QUEUED", "RUNNING"].includes(executionMeta.status)) {
+    status = "running";
+  } else if (executionMeta.status === "INCOMPLETE") {
+    status = "waiting";
   }
-
-  // Determine aggregate startTime, endTime, and totalRunTime
-  const now = Date.now();
-  const startTime = Math.min(
-    ...nodeExecutionResults.map((ner) => ner.add_time.getTime()),
-    now,
-  );
-  const endTime = ["success", "failed"].includes(status)
-    ? Math.max(
-        ...nodeExecutionResults.map((ner) => ner.end_time?.getTime() || 0),
-        startTime,
-      )
-    : now;
-  const duration = (endTime - startTime) / 1000; // Convert to seconds
-  const totalRunTime =
-    nodeExecutionResults.reduce(
-      (cum, node) =>
-        cum +
-        ((node.end_time?.getTime() ?? now) -
-          (node.start_time?.getTime() ?? now)),
-      0,
-    ) / 1000;
-
   return {
-    id: nodeExecutionResults[0].graph_exec_id,
-    graphID: nodeExecutionResults[0].graph_id,
-    graphVersion: nodeExecutionResults[0].graph_version,
+    id: executionMeta.execution_id,
+    graphID: graphMeta.id,
+    graphVersion: graphMeta.version,
     status,
-    startTime,
-    endTime,
-    duration,
-    totalRunTime,
-    nodeExecutionResults: nodeExecutionResults,
-  };
+    startTime: new Date(executionMeta.started_at).getTime(),
+    endTime: executionMeta.ended_at
+      ? new Date(executionMeta.ended_at).getTime()
+      : undefined,
```

Comment:
Parsing date strings from the API into `Date` should happen in the API client. It already does this in `getGraphExecutionInfo` and `stopGraphExecution`.